### DISCUSSION TITLE
Support catchall addresses

### DISCUSSION THREAD

**Discussion by ygoe** [Automattic/gravatar#126] [Ideas] [answered]
I'm using a separate subdomain with a catchall e-mail address for all my service accounts. This allows me to help verify message authenticity, trace back leaked addresses and block single addresses, and escape profiling to some extent. All with zero setup on the mailbox side. It works really well, except that Gravatar won't help me here. Some services rely on Gravatar and can't store profile images themselves. So if I understood it right, I'd need to create an account and add all required e-mail aliases to it. Which can be many and that's a lot of unnecessary work, counteracting the whole catchall idea.

So I'm suggesting a way to register an entire domain and return the same image for all addresses under that domain. For example, the registration could be for `*@services.example.com` and then all addresses like `google@services.example.com` or `github@services.example.com` would be recognised as if I'd have added them manually.

I cannot currently suggest a valid verification method for this. Maybe it could be on the DNS level, or you send the verification message to an undisclosed random (different each time) address under that domain and expect it to be received without prior knowledge.

**Comment by jom** [comment, marked as answer]
Hi @ygoe! As someone who also uses catch-all email addresses, I would love for this to be possible. However, Gravatar's longstanding avatar API makes this impossible as-is. Since we get the hash of the entire email address, we wouldn't be able to parse out what the domain is from the hash. We would have to modify the API to have services integrating with us to also send us a hash of just the email domain in addition to the hash of the email address.

**Comment by aaronfc** [comment]
Thanks for the suggestion.
This has been under our radar for a while now but haven't had the opportunity to work on it yet.

**Comment by theblackhole** [comment]
Hi. I have a similar but slightly different feature suggestion but looking at @jom 's comment, it looks like it has the same technical limitation so I don't know if I should create a new discussion or not :

> **Same avatar for all "+" aliases (email+alias@domain.com)**
> 
> The majority of email providers now supports automatic aliases (e.g. `email+alias1@domain.com` and `email+alias2@domain.com` which automatically redirect to `email@domain.com`).
> They are very useful to organize emails and to track the origin of spams/data leaks.
> 
> Currently, you need to add each alias to gravatar in order to set an avatar because they are treated as separate email addresses.
> I think I would be awesome to have an option to set an avatar to all aliases (and maybe be able to override this option for some aliases? -> bonus feature).

Here the validation wouldn't be at a domain level (since I don't think Gmail and Outlook would appreciate that haha) but the main email address (`email@domain.com`) would validate all "+" aliases.

**Reply by aaronfc** [reply]
Hey @theblackhole, this has the same exact issue that @jom mentioned above. With the current API we only receive a hash of an email, and we can't reverse it to know if it's an alias or not. We would still need for the users to "register" all aliases as secondary emails.
